plugins/couchbase/couchbaseembed.hpp

```diff
@@ -200,21 +206,29 @@ namespace couchbaseembed
             }
         }
 
-        inline void connect()
+        static StringBuffer makeConnectionString(bool useSSL, const char * host, unsigned port, const char * bucketname, const char * connOptions)
         {
-            m_connectionStatus = m_pCouchbaseClient->connect();
-            if (!m_connectionStatus.success())
-                failx("Failed to connect to couchbase instance: %s Reason: %s", m_connectionString.str(), m_connectionStatus.description());
+            StringBuffer connectionString;
```

Comment:
Changed inline function accept a StringBuffer reference and modified callers to accommodate the change.